src/data/sym2.lean

```diff
@@ -453,4 +459,62 @@ end
 
 end decidable
 
+lemma filter_image_quotient_mk [decidable_eq α] (s : finset α) :
+  ((s.product s).image quotient.mk).filter (λ a : sym2 α, ¬a.is_diag) =
+    s.off_diag.image quotient.mk :=
+begin
+  ext ⟨x, y⟩,
+  simp only [mem_image, mem_off_diag, exists_prop, mem_filter, prod.exists, mem_product],
+  split,
+  { rintro ⟨⟨a, b, ⟨ha, hb⟩, h⟩, hab⟩,
+    rw [←h, sym2.is_diag_iff_eq] at hab,
+    exact ⟨a, b, ⟨ha, hb, hab⟩, h⟩ },
+  { rintro ⟨a, b, ⟨ha, hb, hab⟩, h⟩,
+    rw [ne.def, ←sym2.is_diag_iff_eq, h] at hab,
+    exact ⟨⟨a, b, ⟨ha, hb⟩, h⟩, hab⟩ }
+end
+
+lemma prod_quotient_sym2_not_diag [decidable_eq α] (s : finset α) :
```

Comment:
I think this lemma is useful (unsurprisingly) but the statement is a bit hard to parse - maybe it deserves a docstring?

Reply:
I've simplified the statement by moving the first `rw`  to the proof of `card_sym2_not_diag`. What do you think?